src/app/services/note.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Inject, Injectable } from '@angular/core';
+import { Note } from '../models/note';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable, catchError, of, tap } from 'rxjs';
+
+//export c//
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NoteService {
+  idIndex: string = '';
+  httpOptions = {
+    headers: new HttpHeaders({'Content-Type': 'application/json'})
+  };
+  
+  API_URL: string = "https://json-server-vercel-five-chi.vercel.app/";
+  noteCategories: string[] = ['Work', 'School', 'Personal', ]
+  
+  currentNote?: Note;
+  noteList!: Note[];
+
+  constructor(private http: HttpClient) {}
+
+  addNote(): Note {
+    //this.idIndex = Math.random().toString(16).slice(2);
+    console.log(this.idIndex);
+    this.currentNote = {  id: 0,
+      name: '',
+      note: '',
+      category: '',
+    }
+
+    return this.currentNote;
+  }
+
+  getNotes(): Observable<Note[]>{
+    return this.http.get<Note[]>(this.API_URL + "notes", this.httpOptions)
```

Comment:
You can use interpolation for these, makes it a little shorter: \`${this.API_URL}notes\`